test/metabase/test/util/timezone.clj

```diff
@@ -1,25 +1,33 @@
 (ns metabase.test.util.timezone
   (:require [clojure.test :as t]
             [metabase.driver :as driver]
-            [metabase.test.initialize :as initialize])
+            [metabase.test.initialize :as initialize]
+            test-runner)
   (:import java.util.TimeZone))
 
 (defn do-with-system-timezone-id [^String timezone-id thunk]
-  ;; only if the app DB is already set up, we need to make sure plugins are loaded and kill any connection pools that
-  ;; might exist
-  (when (initialize/initialized? :db)
-    (initialize/initialize-if-needed! :plugins)
-    (#'driver/notify-all-databases-updated))
-  (let [original-time-zone       (TimeZone/getDefault)
-        original-system-property (System/getProperty "user.timezone")]
-    (try
-      (TimeZone/setDefault (TimeZone/getTimeZone timezone-id))
-      (System/setProperty "user.timezone" timezone-id)
-      (t/testing (format "JVM timezone set to %s" timezone-id)
-        (thunk))
-      (finally
-        (TimeZone/setDefault original-time-zone)
-        (System/setProperty "user.timezone" original-system-property)))))
+  ;; skip all the property changes if the system timezone doesn't need to be changed.
+  (let [original-timezone        (TimeZone/getDefault)
+        original-system-property (System/getProperty "user.timezone")
+        new-timezone             (TimeZone/getTimeZone timezone-id)]
+    (if (and (= original-timezone new-timezone)
+             (= original-system-property timezone-id))
```

Comment:
some logic to avoid changing stuff if TZ is already set; this lets us use `with-system-timezone-id` inside parallel tests in some cases